src/ws_server.rs

```diff
@@ -30,8 +34,40 @@ impl WsServer {
 
 impl<T: Controller> Handler for WsHandler<T> {
 
-    fn on_open(&mut self, _: Handshake) -> Result<()> {
+    fn on_open(&mut self, handshake: Handshake) -> Result<()> {
         info!("Hello new ws connection");
+
+        let resource = &handshake.request.resource()[..];
+        let url = Url::parse(&format!("http://fox.box{}", resource)).unwrap();
```

Comment:
I added a comment line (good call out).

Unwrapping here shouldn't be an issue since the handshake was already successful and the connection is open for sending and receiving messages. Further, if we panic here only the the thread for that connection would be affected.